p2p/src/sync/peer.rs

```diff
@@ -398,8 +410,17 @@ where
         };
 
         match error {
+            // Due to the fact that p2p is split into several tasks, it is possible to send a
+            // request/response after a peer is disconnected, but before receiving the disconnect
+            // event. Therefore this error can be safely ignored.
+            P2pError::PeerError(PeerError::PeerDoesntExist) => Ok(()),
+            P2pError::MempoolError(
```

Comment:
There has to be a score associated with transactions from the mempool. If this doesn't exist not and it's not feasible, please create a TODO and an issue.